mltensor-explainer.md

```diff
@@ -0,0 +1,421 @@
+# `MLTensor` Explainer
+
+## Authors
+
+- [Austin Sullivan](asully@chromium.org) (Google)
+
+## Participate
+
+- [Issue tracker](https://github.com/webmachinelearning/webnn/issues)
+
+## Introduction
+
+This explainer proposes an `MLTensor` interface which represents a tensor which can be passed as an input and output to `MLGraph` inference.
+
+The machine learning context underlying WebNN may require input and output tensors to be allocated in a specific fashion, such as with a given byte alignment or on a given compute unit (e.g. CPU, GPU, NPU, TPU, etc...). Currently, this requires that the implementation of an `MLGraph` copy in data from the input tensors, execute the graph, and then copy out data from the output tensors.
+
+An `MLTensor` is an opaque tensor which may be created, written to, and read from independently from `MLGraph` inference. Each of these operations is performed on the [timeline](\#timelines) of the associated MLContext, with a clearly defined order of operations. Passing `MLTensor`s as input and output tensors to `MLGraph` inference - as opposed to passing `ArrayBufferView`s as is done today - allows for a decoupling of the uploading/downloading of model inputs/outputs from the model execution itself. This provides several benefits, such as buffer reuse, chained inference, explicit memory management, and the opportunity to interop with WebGPU.
```

Comment:
> We don't require `destroy()` to be called nor is the underlying memory available to the web developer.

Good point. Updated to "explicit destruction" since I think that's more clear than "early" ("early" compared to what?)

Reply:
SGTM.